src/graph_export/embeddings_graph_cli_parser.cpp

```diff
@@ -0,0 +1,142 @@
+//*****************************************************************************
+#include "embeddings_graph_cli_parser.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include "../capi_frontend/server_settings.hpp"
+#include "../ovms_exit_codes.hpp"
+#include "../version.hpp"
+#include "../status.hpp"
+
+namespace ovms {
+
+EmbeddingsGraphSettingsImpl& EmbeddingsGraphCLIParser::defaultGraphSettings() {
+    static EmbeddingsGraphSettingsImpl instance;
+    return instance;
+}
+
+void EmbeddingsGraphCLIParser::createOptions() {
+    this->options = std::make_unique<cxxopts::Options>("ovms --pull [PULL OPTIONS ... ]", "-pull --task embeddings graph options");
+
+    // clang-format off
+    options->add_options("embeddings")
+        ("graph_target_device",
+            "CPU, GPU, NPU or HETERO, default is CPU.",
+            cxxopts::value<std::string>()->default_value("CPU"),
+            "GRAPH_TARGET_DEVICE")
+        ("num_streams",
+            "The number of parallel execution streams to use for the model. Use at least 2 on 2 socket CPU systems.",
+            cxxopts::value<uint32_t>()->default_value("1"),
+            "NUM_STREAMS")
+        ("truncate",
+            "Truncate the prompts to fit to the embeddings model.",
+            cxxopts::value<std::string>()->default_value("false"),
+            "TRUNCATE")
+        ("normalize",
+            "Normalize the embeddings.",
+            cxxopts::value<std::string>()->default_value("false"),
+            "NORMALIZE")
+        ("version",
+            "Version of the model.",
+            cxxopts::value<uint32_t>()->default_value("1"),
+            "VERSION");
+}
+
+void EmbeddingsGraphCLIParser::printHelp() {
+    if (!this->options) {
+        this->createOptions();
+    }
+    std::cout << options->help({"embeddings"}) << std::endl;
+}
+
+void EmbeddingsGraphCLIParser::parse(const std::vector<std::string>& unmatchedOptions) {
+    try {
+        if (!this->options) {
+            this->createOptions();
+        }
+        std::vector<const char*> cStrArray;
+        cStrArray.reserve(unmatchedOptions.size() + 1);
+        cStrArray.push_back("ovms graph");
+        std::transform(unmatchedOptions.begin(), unmatchedOptions.end(), std::back_inserter(cStrArray), [](const std::string& str) { return str.c_str(); });
+        const char* const* args = cStrArray.data();
+        result = std::make_unique<cxxopts::ParseResult>(options->parse(cStrArray.size(), args));
+
+        if (result->unmatched().size()) {
+            std::cerr << "error parsing options - unmatched arguments: ";
+            for (auto& argument : result->unmatched()) {
+                std::cerr << argument << ", ";
+            }
+            std::cerr << std::endl;
+            exit(OVMS_EX_USAGE);
+        }
+    } catch (const std::exception& e) {
+        std::cerr << "error parsing options: " << e.what() << std::endl;
```

Comment:
We already catch that in cli_parser

Reply:
I think its good to have the catch causes per parser.